packages/instrumentation-qdrant/tests/instrumentation.test.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Traceloop
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { context } from "@opentelemetry/api";
+import { AsyncHooksContextManager } from "@opentelemetry/context-async-hooks";
+import { QdrantInstrumentation } from "../src/instrumentation";
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from "@opentelemetry/sdk-trace-base";
+import * as qdrant from "@qdrant/js-client-rest";
+
+const COLLECTION_NAME = "some_collection";
+
+const memoryExporter = new InMemorySpanExporter();
+
+describe("Test Qdrant instrumentation", function () {
+  const provider = new BasicTracerProvider();
+  let instrumentation: QdrantInstrumentation;
+  let contextManager: AsyncHooksContextManager;
+  const qdrantClient: qdrant.QdrantClient = new qdrant.QdrantClient({ url: 'http://127.0.0.1:6333' });
+
+  before(async () => {
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    instrumentation = new QdrantInstrumentation({ traceContent: true });
+    instrumentation.setTracerProvider(provider);
+    instrumentation.manuallyInstrument(qdrant);
+  });
+
+  beforeEach(async function () {
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+    memoryExporter.reset();
+
+
+    qdrantClient.createCollection(COLLECTION_NAME, {
+      vectors: {
+        size: 3,
+        distance: "Cosine",
+      }
+    })
+  });
+
+  afterEach(async () => {
+    await qdrantClient.deleteCollection(COLLECTION_NAME);
+    memoryExporter.reset();
+    context.disable();
+  });
+
+
+  it("should set span attributes for Upsert", async () => {
+    const points =
+    {
+      batch: {
+        ids: [32, 23],
+        vectors: [
+          [1.5, 2.9, 3.4],
+          [9.8, 2.3, 2.9],
+        ],
+        payloads: [{ style: "style3" }, { style: "style4" }],
+      }
+    };
+
+    qdrantClient.upsert(COLLECTION_NAME, points);
+
+    const spans = memoryExporter.getFinishedSpans();
```

Comment:
Hey @nirga. I actually have to debug this.
`memoryExporter.getFinishedSpans()` is returning an empty array.

Reply:
@Anush008 got it! How about instead of running qdrant in CI we record the HTTP calls we're making similar to other tests? We use [polly.js](https://netflix.github.io/pollyjs/) for that

Reply:
We don't need to do that too.
You can use
`QdrantClient(location=":memory:")` for CI purposes.

Reply:
Hi Nir. Did you get a chance to take a look at this?

Reply:
Hey @Anush008 I thought you were trying to fix this so that it will set up qdrant in the CI? The yaml is available on `.github`. Happy to help if you need, but not sure what's the easiest way to run it in CI (are you doing it at Qdrant?)

Reply:
I am still blocked by https://github.com/traceloop/openllmetry-js/pull/278#discussion_r1611341021 actually and was continuing with other parts at work.

Reply:
Great @Anush008 so we can do this here as well! The test passes locally?

Reply:
https://github.com/traceloop/openllmetry-js/pull/278#discussion_r1611341021 blocks the tests too.

Reply:
@Anush008 can you add this (or whatever you're using in your repo) to the CI?

Reply:
https://github.com/traceloop/openllmetry-js/commit/0329f5311837b8686f4142040a51d6c021a51d47

Reply:
Nice @Anush008, so this is ready for review?

Reply:
No. I guess I'm not conveying it right.
https://github.com/traceloop/openllmetry-js/pull/278#discussion_r1611341021 is blocking.